src/js/pages/jobs/JobDetailPage.js

```diff
@@ -0,0 +1,220 @@
+
+    this.state = {
+      currentTab: Object.keys(this.tabs_tabs).shift(),
+      errorCount: 0,
+      isLoading: true
+    };
+  }
+
+  componentDidMount() {
+    super.componentDidMount(...arguments);
+    ChronosStore.monitorJobDetail(this.props.params.id);
+  }
+
+  onChronosStoreJobDetailError() {
+    this.setState({errorCount: this.state.errorCount + 1});
+  }
+
+  onChronosStoreJobDetailChange() {
+    this.setState({errorCount: 0, isLoading: false});
+  }
+
+  handleEditButtonClick() {
+    // TODO: Handle edit button click
+  }
+
+  handleRunNowButtonClick(jobID) {
+    ChronosStore.runJob(jobID);
+  }
+
+  handleMoreDropdownSelection(jobID, selection) {
+    if (selection.id === 'suspend') {
+      ChronosStore.suspendJob(jobID);
+    }
+
+    if (selection.id === 'destroy') {
+      ChronosStore.deleteJob(jobID);
+    }
+  }
+
+  getActionButtons(job) {
+    let jobID = encodeURIComponent(job.getId());
+
+    let dropdownItems = [
+      {
+        className: 'hidden',
+        html: 'More',
+        id: 'more'
+      },
+      {
+        html: 'Suspend',
+        id: 'suspend'
+      },
+      {
+        html: 'Destroy',
+        id: 'destroy'
+      }
+    ];
+
+    return [
+      <button
+        className="button button-inverse button-stroke"
+        key="edit"
+        onClick={this.handleEditButtonClick.bind(this, jobID)}>
+        Edit
+      </button>,
+      <button
+        className="button button-inverse button-stroke"
+        key="run-now"
+        onClick={this.handleRunNowButtonClick.bind(this, jobID)}>
+        Run Now
+      </button>,
+      <Dropdown
+        buttonClassName="dropdown-toggle button button-inverse button-stroke"
+        dropdownMenuClassName="dropdown-menu inverse"
+        dropdownMenuListClassName="dropdown-menu-list"
+        dropdownMenuListItemClassName="clickable"
+        initialID="more"
+        items={dropdownItems}
+        key="more"
+        onItemSelection={this.handleMoreDropdownSelection.bind(this, jobID)}
+        persistentID="more"
+        transition={true}
+        wrapperClassName="dropdown anchor-right" />
+    ];
+  }
+
+  getErrorScreen() {
+    return (
+      <div className="container container-fluid container-pod text-align-center vertical-center inverse">
+        <RequestErrorMsg />
+      </div>
+    );
+  }
+
+  getLoadingScreen() {
+    return (
+      <div className="container container-fluid container-pod text-align-center
+        vertical-center inverse">
+        <div className="row">
+          <div className="ball-scale">
+            <div />
+          </div>
+        </div>
+      </div>
+    );
+  }
+
+  getNavigationTabs() {
+    return (
+      <ul className="tabs tall list-inline flush-bottom inverse">
+        {this.tabs_getUnroutedTabs()}
+      </ul>
+    );
+  }
+
+  getSubTitle(job) {
+    let longestRunningActiveRun = job.getActiveRuns()
+      .getLongestRunningActiveRun();
+    let longestRunningTask = longestRunningActiveRun.getTasks()
```

Comment:
It's not guaranteed that `longestRunningActiveRun` is defined. Please add respective conditions.